go/tasks/plugins/array/outputs.go

```diff
@@ -185,10 +196,11 @@ func AssembleFinalOutputs(ctx context.Context, assemblyQueue OutputAssembler, tC
 			state.GetIndexesToCache(), uint(state.GetOriginalArraySize()))
 
 		err = assemblyQueue.Queue(ctx, workItemID, &outputAssembleItem{
-			varNames:    varNames,
-			finalPhases: finalPhases,
-			outputPaths: tCtx.OutputWriter(),
-			dataStore:   tCtx.DataStore(),
+			varNames:       varNames,
+			finalPhases:    finalPhases,
+			outputPaths:    tCtx.OutputWriter(),
+			dataStore:      tCtx.DataStore(),
+			isAwsSingleJob: taskTemplate.Type == awsBatchTaskType,
```

Comment:
I don't fully understand this, probably because I haven't fully followed this feature. The one-off batch task as defined in the [flytekit pr](https://github.com/flyteorg/flytekit/pull/782) will execute a single job using the AWS batch API and that's why were setting isAwsSingleJob here? we are just reusing the awsbatch array implementation because it already works?

Reply:
yeah, exactly.
we already support running map tasks (array job) on AWS batch, but we can't run regular tasks before.
There are only a few differences:

1. Arrayjob will aggregate output.pb in different directories, but the single job won't. check [here](https://github.com/flyteorg/flyteplugins/pull/228/files#diff-0edeed4ecb7475faa7b319bd5ae0fc578872fb3acd0a20cd0f6cdffcfb914ddeR79-R87)
2. We only need one input reader for a single job instead of a list of readers. check [here](https://github.com/flyteorg/flyteplugins/blob/892ba2cc7401448b9610bfbe5e89507bd5a5cbdd/go/tasks/plugins/array/catalog.go#L65-L97)

Reply:
will update docs in flyte/flytesnack